virtwho/password/__init__.py

```diff
@@ -106,12 +150,12 @@ def _read_or_generate_key_iv(cls):
         iv = hexlify(cls._generate_key())
         try:
             with open(cls.KEYFILE, 'w') as f:
-                f.write("%s\n%s\n" % (key, iv))
+                f.write("%s\n%s\n" % (key.decode(), iv.decode()))
         except IOError as e:
             raise UnwritableKeyFile(str(e))
         os.chmod(cls.KEYFILE, stat.S_IRUSR | stat.S_IWUSR)
         return key, iv
 
     @classmethod
     def _generate_key(cls):
-        return os.urandom(32)
+        return os.urandom(16)
```

Comment:
I suspect 16 is here, because `BLOCKSIZE` is 16 and more random values would be useless. If yes, then I would use `cls.BLOCKSIZE`on all places, where number 16 is used now.